src/Discord.Net.WebSocket/ConnectionManager.cs

```diff
@@ -159,9 +157,9 @@ internal class ConnectionManager
 
             await _onDisconnecting(ex).ConfigureAwait(false);
 
-            await _logger.InfoAsync("Disconnected").ConfigureAwait(false);
-            State = ConnectionState.Disconnected;
             await _disconnectedEvent.InvokeAsync(ex, isReconnecting).ConfigureAwait(false);
+            State = ConnectionState.Disconnected;
```

Comment:
It was moved below to check if the the disconnect deadlock for audio was fixed since async events were expecting a task to be returned which causes deadlocks (depending on the code within the event handler).